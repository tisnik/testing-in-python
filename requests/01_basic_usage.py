#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/get"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# vypis objektu, ktery se vrati
print(response)



# nyni vyzkousime neexistujici endpoint:

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/neexistuje"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# vypis objektu, ktery se vrati
print(response)
