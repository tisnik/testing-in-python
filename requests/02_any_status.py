#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/status/500"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# vypis stavu odpovedi
print(response.status_code)
print(response.ok)
