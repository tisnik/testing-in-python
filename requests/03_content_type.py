#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/get"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# precteni hlavicek
headers = response.headers

# vypis typu internetoveho media
print(headers.get("content-type"))
