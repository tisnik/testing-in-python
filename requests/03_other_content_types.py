#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/status/500"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# precteni hlavicek
headers = response.headers

# vypis typu internetoveho media
print(headers.get("content-type"))

# tento koncový bod vrací obrázek
URL = "http://httpbin.org/image/png"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# precteni hlavicek
headers = response.headers

# vypis typu internetoveho media
print(headers.get("content-type"))
