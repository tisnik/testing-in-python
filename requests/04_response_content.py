#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/get?x=6&y=7&answer=42"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# vypis tela odpovedi
print("Plain text:")
print("-" * 60)
print(response.text)
print("-" * 60)
