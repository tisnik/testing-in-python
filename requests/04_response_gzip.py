#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/gzip"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# vypis tela odpovedi
print("Plain text:")
print("-" * 60)
print(response.text)
print("-" * 60)
