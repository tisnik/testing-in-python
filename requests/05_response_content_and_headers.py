#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/get?x=6&y=7&answer=42"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# precteni hlavicek
headers = response.headers

print("Metadata:")
print("-" * 60)

# vypis typu internetoveho media
print("Typ dat:", headers.get("content-type"))

# vypis delky dat predanych v tele
print("Delka dat:", headers.get("content-length"))

# vypis delky dat predanych v tele
print("Datum:", headers.get("date"))

print("-" * 60)

# vypis tela odpovedi
print("Plain text:")
print("-" * 60)
print(response.text)
print("-" * 60)
