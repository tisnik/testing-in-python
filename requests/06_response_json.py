#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests

# adresa s testovaci REST API sluzbou
URL = "http://httpbin.org/get?x=6&y=7&answer=42"

# poslani HTTP dotazu typu GET
response = requests.get(URL)

# zpracovani odpovedi, ktera prisla ve formatu JSON
data = response.json()

print(data)

args = data["args"]
print(args)

print("x =", args["x"])
print("y =", args["y"])
print("answer =", args["answer"])
