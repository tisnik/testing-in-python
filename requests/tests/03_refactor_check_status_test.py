#!/usr/bin/env python3
# vim: set fileencoding=utf-8

import requests


def expect_ok_response(response):
    assert response is not None
    assert response.ok
    assert response.status_code == 200


def test_get_method_for_existing_endpoint():
    # adresa s testovaci REST API sluzbou
    URL = "http://httpbin.org/get"

    # poslani HTTP dotazu typu GET
    response = requests.get(URL)

    # zakladni test odpovedi
    expect_ok_response(response)


def test_get_method_for_missing_endpoint():
    # adresa s testovaci REST API sluzbou
    URL = "http://httpbin.org/neexistuje"

    # poslani HTTP dotazu typu GET
    response = requests.get(URL)

    # zakladni test odpovedi
    expect_ok_response(response)
