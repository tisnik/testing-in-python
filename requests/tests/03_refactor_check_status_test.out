============================= test session starts ==============================
platform linux -- Python 3.6.6, pytest-5.4.2, py-1.5.2, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/ptisnovs/src/python/testing-in-python/requests/tests
plugins: print-0.1.3, voluptuous-1.0.2
collecting ... collected 2 items

03_refactor_check_status_test.py::test_get_method_for_existing_endpoint PASSED [ 50%]
03_refactor_check_status_test.py::test_get_method_for_missing_endpoint FAILED [100%]

=================================== FAILURES ===================================
_____________________ test_get_method_for_missing_endpoint _____________________

    def test_get_method_for_missing_endpoint():
        # adresa s testovaci REST API sluzbou
        URL = "http://httpbin.org/neexistuje"
    
        # poslani HTTP dotazu typu GET
        response = requests.get(URL)
    
        # zakladni test odpovedi
>       expect_ok_response(response)

03_refactor_check_status_test.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

response = <Response [404]>

    def expect_ok_response(response):
        assert response is not None
>       assert response.ok
E       assert False
E        +  where False = <Response [404]>.ok

03_refactor_check_status_test.py:9: AssertionError
=========================== short test summary info ============================
FAILED 03_refactor_check_status_test.py::test_get_method_for_missing_endpoint
========================= 1 failed, 1 passed in 0.88s ==========================
